L1Trigger/TrackFindingTracklet/plugins/ProducerKFout.cc

```diff
@@ -0,0 +1,150 @@
+  };
+
+  ProducerKFout::ProducerKFout(const ParameterSet& iConfig) :
+    iConfig_(iConfig)
+  {
+    const string& label = iConfig.getParameter<string>("LabelKF");
+    const string& branchStubs = iConfig.getParameter<string>("BranchAcceptedStubs");
+    const string& branchTracks = iConfig.getParameter<string>("BranchAcceptedTracks");
+    // book in- and output ED products
+    edGetTokenStubs_ = consumes<TTDTC::Streams>(InputTag(label, branchStubs));
+    edGetTokenTracks_ = consumes<StreamsTrack>(InputTag(label, branchTracks));
+    edPutToken_ = produces<TTTracks>(branchTracks);
+    // book ES products
+    esGetTokenSetup_ = esConsumes<Setup, SetupRcd, Transition::BeginRun>();
+    esGetTokenDataFormats_ = esConsumes<DataFormats, DataFormatsRcd, Transition::BeginRun>();
+    // initial ES products
+    setup_ = nullptr;
+    dataFormats_ = nullptr;
+    // used data formats
+    zT_ = nullptr;
+    cot_ = nullptr;
+    phi_ = nullptr;
+    z_ = nullptr;
+  }
+
+  void ProducerKFout::beginRun(const Run& iRun, const EventSetup& iSetup) {
+    // helper class to store configurations
+    setup_ = &iSetup.getData(esGetTokenSetup_);
+    if (!setup_->configurationSupported())
+      return;
+    // check process history if desired
+    if (iConfig_.getParameter<bool>("CheckHistory"))
+      setup_->checkHistory(iRun.processHistory());
+    // helper class to extract structured data from TTDTC::Frames
+    dataFormats_ = &iSetup.getData(esGetTokenDataFormats_);
+    // used data formats
+    zT_ = &dataFormats_->format(Variable::zT, Process::kfin);
+    cot_ = &dataFormats_->format(Variable::cot, Process::kfin);
+    phi_ = &dataFormats_->format(Variable::phi, Process::sf);
+    z_ = &dataFormats_->format(Variable::z, Process::sf);
+  }
+
+  void ProducerKFout::produce(Event& iEvent, const EventSetup& iSetup) {
+    // empty KFout product
+    TTTracks ttTracks;
+    // read in KF Product and produce KFout product
+    if (setup_->configurationSupported()) {
+      Handle<TTDTC::Streams> handleStubs;
+      iEvent.getByToken<TTDTC::Streams>(edGetTokenStubs_, handleStubs);
+      const TTDTC::Streams& streamsStubs = *handleStubs.product();
+      Handle<StreamsTrack> handleTracks;
+      iEvent.getByToken<StreamsTrack>(edGetTokenTracks_, handleTracks);
+      const StreamsTrack& streamsTracks = *handleTracks.product();
+      // count number of kf tracks
+      int nTracks(0);
+      for (const StreamTrack& stream : streamsTracks)
+        nTracks += accumulate(stream.begin(), stream.end(), 0, [](int& sum, const FrameTrack& frame){ return sum += frame.first.isNonnull() ? 1 : 0; });
+      ttTracks.reserve(nTracks);
+      // convert kf track frames per region and stub frames per region and layer to TTTracks
+      for (int region = 0; region < setup_->numRegions(); region++) {
+        const int offset = region * setup_->numLayers();
+        int iTrk(0);
+        for (const FrameTrack& frameTrack : streamsTracks[region]) {
```

Comment:
It seems like the maximum allowed region is 9 (setup_->numRegions()=9) but there are 18 items in this "streamsTracks" container. This loop basically ignores tracks found in region 9-17 ?

Reply:
If you run with this variable unchanged you get half the number of tracks you would expect and only half the phi sectors are filled, changing the number of allowed regions to 18 you recover these lost tracks

Reply:
Upsi, is fixed.